config/application.rb

```diff
@@ -47,9 +38,80 @@ class Application < Rails::Application
     # Add Yarn node_modules folder to the asset load path.
     config.assets.paths << Rails.root.join('node_modules')
 
-    # Configure responders gem flash keys.
-    # NOTE: This didn't work when put in config/initializers/responders.rb.
-    config.responders.flash_keys = [:success, :error]
+    # Settings below are configurable
+
+    # Set the timezone
+    config.time_zone = Settings.rails.time_zone
+
+    # Use Resque for background jobs
+    config.active_job.queue_adapter = Settings.rails.active_job.queue_adapter.to_sym
+
+    # Markus Session Store configuration
+    # Be sure to restart your server when you modify this part.
+    #
+    # Your secret key for verifying cookie session data integrity.
+    # If you change this key, all old sessions will become invalid!
+    # Make sure the secret is at least 30 characters and all random,
+    # no regular words or you'll be exposed to dictionary attacks.
+    # Please make sure :_key is named uniquely if you are hosting
+    # several MarkUs instances on one machine. Also, make sure you change
+    # the :secret string to something else than you find below.
+
+    Rails.application.config.session_store(
+      Settings.rails.session_store.type.to_sym,
+      key: Settings.rails.session_store.args.key,
+      path: Settings.rails.session_store.args.path,
+      expire_after: Settings.rails.session_store.args.expire_after.days,
+      secure: Settings.rails.session_store.args.secure,
+      same_site: Settings.rails.session_store.args.same_site.to_sym
+    )
+
+    # Email notifications
+    config.action_mailer.delivery_method = Settings.rails.action_mailer.delivery_method.to_sym
+    config.action_mailer.smtp_settings = Settings.rails.action_mailer.smtp_settings.to_h
```

Comment:
This is interesting, why is the `to_h` necessary here? (What's the value of `Settings.rails.action_mailer.smtp_settings`?)

Reply:
`config.action_mailer.smtp_settings` requires a hash object. `Settings.rails.action_mailer.smtp_settings` is either a `Config` struct or `nil`. In either case we end up with a properly formatted hash (`nil.to_h  == {}`)